Fix crash when multiple `# fmt: skip` comments are used in multi-part if-clause (#4731)

Nikhil172913832 opened the pull request:
<!-- Hello! Thanks for submitting a PR. To help make things go a bit smoother
     we would appreciate that you go through this template. -->

### Description

Closes #4731
Closes #3608
Closes #4873 

Black was crashing with a parsing error when multiple `# fmt: skip` comments appeared in multi-part if-clauses. The crash occurred because when the second `# fmt: skip` was processed, its `prev_sibling` could include a `STANDALONE_COMMENT` node created by the first `# fmt: skip`. Since these synthetic nodes don't include trailing newlines (those are added later in the formatting pipeline), when the nodes were joined together via `str()`, they'd concatenate on the same line, producing invalid Python syntax that couldn't be parsed.

The fix ensures that any `STANDALONE_COMMENT` nodes in `ignored_nodes` have trailing newlines before being stringified. This is done right before the join operation in `_handle_regular_fmt_block()`. The approach is similar to what was suggested in the issue, but processes all nodes in `ignored_nodes` (not just the first) and includes a check to avoid adding duplicate newlines.

The existing cleanup logic that strips trailing newlines from `hidden_value` if it ends with a NEWLINE leaf continues to work as expected.

### Checklist - did you ...

<!-- If any of the following items aren't relevant for your contribution,
     please still tick them so we know you've gone through the checklist.

     - Please familiarize yourself with Black's stability policy, linked
       below. Code style changes are only allowed under the `--preview` flag
       until maintainers move them to stable in the next calendar year.
     - All user-facing changes should get a changelog entry. If this isn't
       user-facing, signal to us that this should get the magical label to
       silence the check.
     - Tests are required for all bugfixes and new features.
     - Documentation changes are necessary for most formatting changes and
       other enhancements. -->

- [x] Implement any code style changes under the `--preview` style, following the
      stability policy?
- [x] Add an entry in `CHANGES.md` if necessary?
- [x] Add / update tests if necessary?
- [x] Add new / update outdated documentation?

<!-- Just as a reminder, everyone in all psf/black spaces, including PRs, must
     follow the PSF Code of Conduct (link below).

     Finally, thanks once again for your time and effort. If you have any
     feedback regarding your experience contributing here, please let us know!

     Helpful links:

     - PSF COC: https://www.python.org/psf/conduct/
     - Contributing docs: https://black.readthedocs.io/en/latest/contributing/index.html
     - Chat on Python Discord: https://discord.gg/RtVdv86PrH
     - Stability policy: https://black.readthedocs.io/en/latest/the_black_code_style/index.html -->

cobaltt7 commented:
Thanks for the contribution! I've tested, and it seems that this change also fixes #3608 - could you add test cases for that? Code looks good otherwise.

Nikhil172913832 commented:
I’ve added test cases for both issues and updated my approach to address #4873, which was causing a timeout locally. Let me know if any further edits are required.
